11-payment-encoding.md

```diff
@@ -145,12 +145,14 @@ Currently defined tagged fields are:
 * `x` (6): `data_length` variable. `expiry` time in seconds (big-endian). Default is 3600 (1 hour) if not specified.
 * `c` (24): `data_length` variable. `min_final_cltv_expiry` to use for the last HTLC in the route. Default is 18 if not specified.
 * `f` (9): `data_length` variable, depending on version. Fallback on-chain address: for Bitcoin, this starts with a 5-bit `version` and contains a witness program or P2PKH or P2SH address.
-* `r` (3): `data_length` variable. One or more entries containing extra routing information for a private route; there may be more than one `r` field
-   * `pubkey` (264 bits)
-   * `short_channel_id` (64 bits)
-   * `fee_base_msat` (32 bits, big-endian)
-   * `fee_proportional_millionths` (32 bits, big-endian)
-   * `cltv_expiry_delta` (16 bits, big-endian)
+* `r` (3): `data_length` variable. One or more entries containing extra routing information for a private route; there may be more than one `r` field.
+   * Fields are
+      * `pubkey` (264 bits)
+      * `short_channel_id` (64 bits)
+      * `fee_base_msat` (32 bits, big-endian)
+      * `fee_proportional_millionths` (32 bits, big-endian)
+      * `cltv_expiry_delta` (16 bits, big-endian)
+   * Be aware that `data_length` can express `2^10` (=`1024`) as the longest, and a sigle hop will be 51 bytes (= 408 bits). So if you want to express the hint with more than three hops, you must use two `r` fields.
```

Comment:
This is actually wrong; you can't use two r fields. That would be separate hints.  You simply can't express more than 3 hops :(